### PR TITLE
Add GoogleAuthenticatorInterface

### PR DESCRIPTION
**Pull request by umpirsky**
Addresses testing issues discussed in https://github.com/sonata-project/GoogleAuthenticator/issues/110.

### Added
- Added `GoogleAuthenticatorInterface`

### CODE REVIEW

File: src/GoogleAuthenticatorInterface.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
[the remark this answers is not in the document]
Comment (umpirsky) [reply src/GoogleAuthenticatorInterface.php]: I'd keep them only in interface and use `@inheritdoc` in `GoogleAuthenticator`?

[the remark this answers is not in the document]
Comment (kunicmarko20) [reply src/GoogleAuthenticatorInterface.php]: We use MIT everywhere, I think this one is a mistake

File: src/GoogleAuthenticatorInterface.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Sonata\GoogleAuthenticator;
+
+interface GoogleAuthenticatorInterface
+{
+    /**
+     * @param string $secret
+     * @param string $code
+     *
+     * @return bool
```
Comment (kunicmarko20) [inline src/GoogleAuthenticatorInterface.php]: We already have a return type, this is not needed, can you remove it, please.

File: src/GoogleAuthenticatorInterface.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+     */
+    public function checkCode($secret, $code): bool;
+
+    /**
+     * NEXT_MAJOR: add the interface typehint to $time and remove deprecation.
+     *
+     * @param string                                   $secret
+     * @param float|string|int|null|\DateTimeInterface $time
+     *
+     * @return string
```
Comment (kunicmarko20) [inline src/GoogleAuthenticatorInterface.php]: We already have a return type, this is not needed, can you remove it, please.

File: src/GoogleAuthenticatorInterface.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public function getCode($secret, /* \DateTimeInterface */$time = null): string;
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @param string $user
+     * @param string $hostname
+     * @param string $secret
+     *
+     * @return string
```
Comment (kunicmarko20) [inline src/GoogleAuthenticatorInterface.php]: We already have a return type, this is not needed, can you remove it, please.

File: src/GoogleAuthenticatorInterface.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+     *
+     * @deprecated deprecated as of 2.1 and will be removed in 3.0. Use Sonata\GoogleAuthenticator\GoogleQrUrl::generate() instead.
+     */
+    public function getUrl($user, $hostname, $secret): string;
+
+    /**
+     * @return string
```
Comment (kunicmarko20) [inline src/GoogleAuthenticatorInterface.php]: We already have a return type, this is not needed, can you remove it, please.

File: src/GoogleAuthenticatorInterface.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (umpirsky) [reply src/GoogleAuthenticatorInterface.php]: It's copy pasta from https://github.com/sonata-project/GoogleAuthenticator/blob/2.x/src/GoogleAuthenticator.php#L15

**Comment by greg0ire** [comment]
Thanks @umpirsky !

File: src/GoogleAuthenticatorInterface.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * NEXT_MAJOR: Remove this method.
+     *
+     * @param string $user
+     * @param string $hostname
+     * @param string $secret
+     *
+     * @deprecated deprecated as of 2.1 and will be removed in 3.0. Use Sonata\GoogleAuthenticator\GoogleQrUrl::generate() instead.
+     */
```
Comment (core23) [inline src/GoogleAuthenticatorInterface.php]: Why did you add this to the interface?

File: src/GoogleAuthenticatorInterface.php [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Sonata\GoogleAuthenticator;
+
+interface GoogleAuthenticatorInterface
+{
+    /**
+     * @param string $secret
+     * @param string $code
+     */
+    public function checkCode($secret, $code): bool;
```
Comment (core23) [inline src/GoogleAuthenticatorInterface.php]: you could remove the PHPdoc from the impl class

File: src/GoogleAuthenticatorInterface.php (hunk 6 above)
Comment (umpirsky) [reply src/GoogleAuthenticatorInterface.php]: Well, that's true, will remove it.

**Comment by OskarStark** [comment]
Released [2.2.0](https://github.com/sonata-project/GoogleAuthenticator/releases/tag/2.2.0)
